File: test/claude-config.test.mjs
import { describe, it } from "node:test";
import assert from "node:assert/strict";
import path from "node:path";
import {
  resolveClaudeConfigDir,
  settingsPath,
} from "../src/claude-config.mjs";

describe("resolveClaudeConfigDir", () => {
  const home = "/home/alice";

  it("defaults to ~/.claude when CLAUDE_CONFIG_DIR unset", () => {
    assert.equal(
      resolveClaudeConfigDir({}, home),
      path.join(home, ".claude"),
    );
  });

  it("uses CLAUDE_CONFIG_DIR when set", () => {
    assert.equal(
      resolveClaudeConfigDir({ CLAUDE_CONFIG_DIR: "/custom/claude" }, home),
      path.normalize("/custom/claude"),
    );
  });

  it("treats empty CLAUDE_CONFIG_DIR as unset", () => {
    assert.equal(
      resolveClaudeConfigDir({ CLAUDE_CONFIG_DIR: "" }, home),
      path.join(home, ".claude"),
    );
  });

  it("treats whitespace-only CLAUDE_CONFIG_DIR as unset", () => {
    assert.equal(
      resolveClaudeConfigDir({ CLAUDE_CONFIG_DIR: "   " }, home),
      path.join(home, ".claude"),
    );
  });

  it("configDirOverride wins over env", () => {
    assert.equal(
      resolveClaudeConfigDir(
        { CLAUDE_CONFIG_DIR: "/from/env" },
        home,
        "/from/override",
      ),
      path.normalize("/from/override"),
    );
  });

  it("empty configDirOverride falls through to env", () => {
    assert.equal(
      resolveClaudeConfigDir(
        { CLAUDE_CONFIG_DIR: "/from/env" },
        home,
        "",
      ),
      path.normalize("/from/env"),
    );
  });
});

describe("settingsPath", () => {
  it("appends settings.json to resolved config dir", () => {
    const home = "/home/bob";
    assert.equal(
      settingsPath({}, home),
      path.join(home, ".claude", "settings.json"),
    );
    assert.equal(
      settingsPath({ CLAUDE_CONFIG_DIR: "/x" }, home),
      path.join(path.normalize("/x"), "settings.json"),
    );
  });
});

File: shared/src/commonTest/kotlin/com/cachefix/gui/ClaudeConfigTest.kt
package com.cachefix.gui

import com.cachefix.gui.config.ClaudeConfig
import com.cachefix.gui.path.Paths
import kotlin.test.Test
import kotlin.test.assertEquals

class ClaudeConfigTest {
    private val home = "/home/alice"

    @Test
    fun defaultsToDotClaudeWhenEnvUnset() {
        assertEquals(
            Paths.join(home, ".claude"),
            ClaudeConfig.resolveClaudeConfigDir(emptyMap(), home),
        )
    }

    @Test
    fun usesClaudeConfigDirWhenSet() {
        assertEquals(
            Paths.normalize("/custom/claude"),
            ClaudeConfig.resolveClaudeConfigDir(mapOf("CLAUDE_CONFIG_DIR" to "/custom/claude"), home),
        )
    }

    @Test
    fun treatsEmptyClaudeConfigDirAsUnset() {
        assertEquals(
            Paths.join(home, ".claude"),
            ClaudeConfig.resolveClaudeConfigDir(mapOf("CLAUDE_CONFIG_DIR" to ""), home),
        )
    }

    @Test
    fun treatsWhitespaceOnlyAsUnset() {
        assertEquals(
            Paths.join(home, ".claude"),
            ClaudeConfig.resolveClaudeConfigDir(mapOf("CLAUDE_CONFIG_DIR" to "   "), home),
        )
    }

    @Test
    fun configDirOverrideWinsOverEnv() {
        assertEquals(
            Paths.normalize("/from/override"),
            ClaudeConfig.resolveClaudeConfigDir(
                mapOf("CLAUDE_CONFIG_DIR" to "/from/env"),
                home,
                "/from/override",
            ),
        )
    }

    @Test
    fun emptyOverrideFallsThroughToEnv() {
        assertEquals(
            Paths.normalize("/from/env"),
            ClaudeConfig.resolveClaudeConfigDir(
                mapOf("CLAUDE_CONFIG_DIR" to "/from/env"),
                home,
                "",
            ),
        )
    }

    @Test
    fun settingsPathAppendsSettingsJson() {
        val bob = "/home/bob"
        assertEquals(
            Paths.join(bob, ".claude", "settings.json"),
            ClaudeConfig.settingsPath(emptyMap(), bob),
        )
        assertEquals(
            Paths.join(Paths.normalize("/x"), "settings.json"),
            ClaudeConfig.settingsPath(mapOf("CLAUDE_CONFIG_DIR" to "/x"), bob),
        )
    }
}
